nni/retiarii/execution/base.py

```diff
@@ -36,7 +37,7 @@ class BaseExecutionEngine(AbstractExecutionEngine):
     Resource management is implemented in this class.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, devices=None) -> None:
```

Comment:
Who will use this `devices`? If it's only useful for `CGOExecutionEngine`, suggest putting it there.

Reply:
Moved devices to CGOExecutionEngine.